### PR TITLE
homework 2-1

### PR DESCRIPTION
**Pull request by gerganaivvanova**

### CODE REVIEW

**Review by yll090 — APPROVED** [review]

File: swiftui-project/View/Layouts/ProductRating.swift [hunk 1]
```diff
@@ -10,19 +10,19 @@ import SwiftUI
 
 struct Rating: View {
     var rating: Int
-    let maxRating: Int = 5
     
     var body: some View {
         HStack(spacing: 2) {
```
Comment (gbuyukliev-accedia) [inline swiftui-project/View/Layouts/ProductRating.swift:15]: Should export value as a constant.

File: swiftui-project/View/Login/LoginView.swift [hunk 2]
```diff
@@ -21,72 +21,76 @@ struct LoginView: View {
...
                
                 Button(action:loginModel.login){
                         Text("Log in")
-                            .foregroundColor(.white)
-                            .padding(.vertical, 10)
-                            .padding(.horizontal, 50)
-                            .background(loginModel.isLogInButtonDisabled() ? .gray : .blue)
-                            .cornerRadius(5.0)
-                            .padding(.top, 50)
+                            .foregroundColor(Color("Base Color"))
```
Comment (gbuyukliev-accedia) [inline swiftui-project/View/Login/LoginView.swift]: A little weird naming. Better call it `BaseColor` instead. Spacings in naming are generally a bit risky.

File: swiftui-project/View/Login/LoginView.swift [hunk 3]
```diff
@@ -21,72 +21,76 @@ struct LoginView: View {
...
+                            .font(.callout)
+                            .padding(.vertical, UIConstants.basePadding)
+                            .frame(maxWidth: .infinity)
+                            .background(loginModel.isLogInButtonDisabled() ? Color("Disabled Color") : Color("Textcolor"))
```
Comment (gbuyukliev-accedia) [inline swiftui-project/View/Login/LoginView.swift]: `DisabledColor`
`TextColor`

File: swiftui-project/View/Login/LoginView.swift [hunk 4]
```diff
@@ -21,72 +21,76 @@ struct LoginView: View {
...
+                            .cornerRadius(UIConstants.mediumCornerRadius)
+                            .padding(.top, UIConstants.topBottomPadding)
                     }
                 .disabled(loginModel.isLogInButtonDisabled() || !self.isEmailValid)
                 .navigationDestination(isPresented: $loginModel.isLoginValid, destination: {ProductDetailsView()})
             }
             
             .frame(maxWidth: .infinity,maxHeight: .infinity)
-            .padding(.horizontal,26)
+            .padding(.horizontal,UIConstants.horizontalPadding)
```
Comment (gbuyukliev-accedia) [inline swiftui-project/View/Login/LoginView.swift]: `horizontalPadding` doesn't really sound reusable.

Better naming might be `smallPadding`, `padding`, `bigPadding`, `largePadding`, etc. so they have more applications.

File: swiftui-project/View/ProductDetails/ProductDetailsView.swift [hunk 5]
```diff
@@ -50,26 +50,26 @@ struct ProductDetailsView: View {
...
-                        .padding(10)
-                        .background(.white)
-                        .cornerRadius(26)
-                        .padding(.vertical,15)
-                        .padding(.horizontal, 30)
+                        .padding(UIConstants.basePadding)
+                        .background(Color("Base Color"))
+                        .cornerRadius(UIConstants.mediumCornerRadius)
+                        .padding(.vertical, UIConstants.basePadding)
+                        .padding(.horizontal, UIConstants.horizontalPadding) //30
```
Comment (gbuyukliev-accedia) [inline swiftui-project/View/ProductDetails/ProductDetailsView.swift]: `//30` ?

File: swiftui-project/View/ProductDetails/ProductDetailsView.swift [hunk 6]
```diff
@@ -79,35 +79,39 @@ struct ProductDetailsView: View {
...
+                                        }
+                .padding()
             }
-                Button(action: {}, label: {
+            
+            Button(action: {}, label: {
                     Text("Add to cart")
                         .foregroundColor(.white)
-                        .frame(width: 340, height: 46)
+                        //.frame(width: 340, height: 46)
```
Comment (gbuyukliev-accedia) [inline swiftui-project/View/ProductDetails/ProductDetailsView.swift]: Left out comment.

File: swiftui-project/View/ProductDetails/ProductDetailsView.swift [hunk 7]
```diff
@@ -50,26 +50,26 @@ struct ProductDetailsView: View {
...
+                        .padding(UIConstants.basePadding)
+                        .background(Color("Base Color"))
+                        .cornerRadius(UIConstants.mediumCornerRadius)
+                        .padding(.vertical, UIConstants.basePadding)
+                        .padding(.horizontal, UIConstants.horizontalPadding) //30
                     })
                 }
                 
                 
                 VStack(alignment: .leading) {
```
Comment (gbuyukliev-accedia) [inline swiftui-project/View/ProductDetails/ProductDetailsView.swift]: Mind you that without implicitly giving a value to `spacing`, even a `0`, the `Stack` will introduce "magical" spacings by default and you design won't be properly aligned.

**Review by gbuyukliev-accedia — APPROVED** [review]

File: swiftui-project/View/Login/LoginView.swift (hunk 3 above)
Comment (gerganaivvanova) [reply swiftui-project/View/Login/LoginView.swift]: All asset names are changed.

File: swiftui-project/View/ProductDetails/ProductDetailsView.swift (hunk 5 above)
Comment (gerganaivvanova) [reply swiftui-project/View/ProductDetails/ProductDetailsView.swift]: removed

File: swiftui-project/View/ProductDetails/ProductDetailsView.swift (hunk 7 above)
Comment (gerganaivvanova) [reply swiftui-project/View/ProductDetails/ProductDetailsView.swift]: Spacing added

File: swiftui-project/View/ProductDetails/ProductDetailsView.swift (hunk 6 above)
Comment (gerganaivvanova) [reply swiftui-project/View/ProductDetails/ProductDetailsView.swift]: Removed

File: swiftui-project/View/Login/LoginView.swift (hunk 4 above)
Comment (gerganaivvanova) [reply swiftui-project/View/Login/LoginView.swift]: Changed

File: swiftui-project/View/Login/LoginView.swift (hunk 2 above)
Comment (gerganaivvanova) [reply swiftui-project/View/Login/LoginView.swift]: Changed

**Pull request merged**